hack/boilerplate/boilerplate.py

```diff
@@ -16,22 +16,54 @@
 
 from __future__ import print_function
 
+import argparse
 import json
 import mmap
 import os
 import re
 import sys
 
-def PrintError(*err):
-  print(*err, file=sys.stderr)
+parser = argparse.ArgumentParser()
+parser.add_argument("filenames", help="list of files to check, all files if unspecified", nargs='*')
+args = parser.parse_args()
+
+rootdir = os.path.dirname(__file__) + "/../../"
+rootdir = os.path.abspath(rootdir)
+
+def get_refs():
+    extRE = re.compile("boilerplate.([^.]+).txt")
+
+    refs = {}
+    for root, dirs, files in os.walk(os.path.join(rootdir, "hack/boilerplate/")):
+        for pathname in files:
+            match = extRE.match(pathname)
+            if not match:
+                continue
+            extension = match.group(1)
+
+            ref_filename = os.path.join(root, pathname)
+            ref_file = open(ref_filename, 'r')
+            ref = ref_file.read().splitlines()
+            ref_file.close()
+            refs[extension] = ref
+
+    return refs
+
+def file_passes(filename, refs, regexs):
+    # This horrific line gets the go or py or .whatever extension from a given pathname
+    extension = os.path.splitext(filename)[-1].strip().lower().split(".")[-1]
```

Comment:
Is there really whitespace that needs to be stripped?

Also `os.path.splitext` always returns a 2 element tuple, so the negative index could just be `[1]`. And instead of splitting on the period, you could just slice it off.

```python
extention = os.path.splitext(filename)[1][1:]
```